// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: list_credentials.proto

package identityd2

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// define the regex for a UUID once up-front
var _list_credentials_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on ListCredentialsRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListCredentialsRequest) Validate() error {
	if m == nil {
		return nil
	}

	if v, ok := interface{}(m.GetCredential()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListCredentialsRequestValidationError{
				field:  "Credential",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// ListCredentialsRequestValidationError is the validation error returned by
// ListCredentialsRequest.Validate if the designated constraints aren't met.
type ListCredentialsRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListCredentialsRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListCredentialsRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListCredentialsRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListCredentialsRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListCredentialsRequestValidationError) ErrorName() string {
	return "ListCredentialsRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListCredentialsRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListCredentialsRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListCredentialsRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListCredentialsRequestValidationError{}

// Validate checks the field values on ListCredentialsResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListCredentialsResponse) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetCredentials() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListCredentialsResponseValidationError{
					field:  fmt.Sprintf("Credentials[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// ListCredentialsResponseValidationError is the validation error returned by
// ListCredentialsResponse.Validate if the designated constraints aren't met.
type ListCredentialsResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListCredentialsResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListCredentialsResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListCredentialsResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListCredentialsResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListCredentialsResponseValidationError) ErrorName() string {
	return "ListCredentialsResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListCredentialsResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListCredentialsResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListCredentialsResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListCredentialsResponseValidationError{}
